custom_components/aqara_gateway/__init__.py

```diff
@@ -70,12 +70,6 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     await hass.data[DOMAIN][entry.entry_id].async_connect()
 
-    async def async_stop_mqtt(_event: Event):
-        """Stop MQTT component."""
-        await hass.data[DOMAIN][entry.entry_id].async_disconnect()
-
-    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_stop_mqtt)
-
```

Comment:
Not necessary here because home assistant calls `async_unload_entry` when being stopped.